### PR TITLE
metrics: add fail reason

### PR DESCRIPTION
**Pull request by BusyJay**
So we can know what's happening under the hook more quickly.

@siddontang @disksing PTAL

### CODE REVIEW

File: src/storage/engine/raftkv.rs [hunk 1]
```diff
@@ -246,25 +288,32 @@ impl<S: RaftStoreRouter> Engine for RaftKv<S> {
...
+                        cb((cb_ctx, Err(e)))
+                    }
                 }
-            }
-        }));
-        Ok(())
+            })
+            .map_err(|e| {
+                let tag = get_tag_from_error(&e);
+                ASYNC_REQUESTS_COUNTER_VEC.with_label_values(&["write", tag]).inc();
```
Comment (siddontang) [inline src/storage/engine/raftkv.rs]: why `write` here?

Comment (BusyJay) [inline src/storage/engine/raftkv.rs]: `write` indicates it's a write command.

Comment (siddontang) [inline src/storage/engine/raftkv.rs]: why `write` here?

Comment (siddontang) [inline src/storage/engine/raftkv.rs]: sorry, I mean the following `write`.

Comment (disksing) [inline src/storage/engine/raftkv.rs]: I think @siddontang is trying to identify this "write" should be "snapshot"...

Comment (siddontang) [inline src/storage/engine/raftkv.rs]: tag shoud be snapshot

**Comment by disksing** [comment]
LGTM

**Pull request merged**